tm2py/components/demand/commercial.py

```diff
@@ -0,0 +1,611 @@
+            if spec["use_k_factors"]:
+                friction_matrix = friction_matrix * k_factors
+            # run matrix balancing
+            prod_attr_matrix = self._matrix_balancing(
+                friction_matrix,
+                trip_ends[f"{name}_prod"].to_numpy(),
+                trip_ends[f"{name}_attr"].to_numpy(),
+                name,
+            )
+            daily_demand[name] = (
+                0.5 * prod_attr_matrix + 0.5 * prod_attr_matrix.transpose()
+            )
+            self.logger.log(
+                f"{name}, prod sum: {prod_attr_matrix.sum()}, "
+                f"daily sum: {daily_demand[name].sum()}",
+                level="DEBUG",
+            )
+        return daily_demand
+
+    def _get_blended_time(self, name: str) -> NumpyArray:
+        """Blend AM and MD truck times for distribution calculations.
+
+        Uses 1/3 * AM + 2/3 * MD, sources skims from generated OMX files
+        """
+        skim_path_tmplt = self.get_abs_path(self.config.highway.output_skim_path)
+        with OMXManager(skim_path_tmplt.format(period="AM"), "r") as am_file:
+            am_time = am_file.read(f"am_{name}_time")
+        with OMXManager(skim_path_tmplt.format(period="MD"), "r") as md_file:
+            md_time = md_file.read(f"md_{name}_time")
+        # blend truck times
+        # compute blended truck time as an average 1/3 AM and 2/3 MD
+        #     NOTE: Cube outputs skims\COM_HWYSKIMAM_taz.tpp, skims\COM_HWYSKIMMD_taz.tpp
+        #           are in the highway_skims_{period}.omx files in Emme version
+        #           with updated matrix names, {period}_trk_time, {period}_lrgtrk_time.
+        #           Also, there will no longer be separate very small, small and medium
+        #           truck times, as they are assigned together as the same class.
+        #           There is only the trk_time.
+        blend_time = 1 / 3.0 * am_time + 2 / 3.0 * md_time
+        return blend_time
+
+    @LogStartEnd(level="DEBUG")
+    def _time_of_day(
+        self, daily_demand: Dict[str, Dict[str, NumpyArray]]
+    ) -> Dict[str, NumpyArray]:
+        """Apply period factors to convert daily demand totals to per-period demands.
+
+        Args:
+            daily_demand: dictionary of truck type name to numpy array of
+                truck type daily demand
+
+        Returns:
+             Nested dictionary of period name to truck type name to numpy array
+             of per period demand.
+        """
+        period_demand = {}
+        for period in self.time_period_names():
+            factor_map = _time_of_day_split[period]
+            demand = {}
+            for name, factor in factor_map.items():
+                demand[name] = np.around(factor * daily_demand[name], decimals=2)
+            period_demand[period] = demand
+        return period_demand
+
+    @LogStartEnd(level="DEBUG")
+    def _toll_choice(
```

Comment:
why is this here but here is another one in emme/matrix?

If this is common function, should pull out.